Source/WebCore/platform/graphics/BitmapImageSource.cpp

```diff
@@ -243,6 +243,11 @@ EncodedDataStatus BitmapImageSource::setData(FragmentedSharedBuffer* data, bool
     if (!data)
         return EncodedDataStatus::Unknown;
 
+#if USE(CG)
+    // FIXME: Usually this is not needed. But currently this is the only way to get "correct" gain map data from ImageIO.
+    m_decoder = nullptr;
```

Comment:
Does this imply that we fully re-decode on every incremental data chunk? What are the performance implications?

Reply:
The documentation of `CGImageSourceUpdateData()` says:
```
/* Update the incremental image source `isrc' with new data.  The new data
 * must include all the previous data plus any additional new data. The
 * `final' parameter should be true when the final set of data is provided;
 * false otherwise. */
```
So I am not sure whether CG/ImageIO keeps the `CGImageSource` decoded metadata while calling `CGImageSourceUpdateData()` multiple times. If it they do, that means the CGImageSource metadata will to be recoded with every incremental update.

I do not know right now the perf implications of recreating the `CGImageSource` but I will investigate.

Reply:
I turned it this is a bug below WebKit. I deleted this line and I filed rdar://149562544 to track it internally. So no perf testing is needed for now.